### PR TITLE
add: openai moderation

### PR DESCRIPTION
**Pull request by Justin-ZL**

### CODE REVIEW

File: metagpt/provider/openai_api.py [hunk 1]
```diff
@@ -286,3 +295,17 @@ def get_max_tokens(self, messages: list[dict]):
...
+
+    def moderation(self, content: Union[str, List[str]]):
+        try:
+            if not content:
+                logger.error("content cannot be empty!")
+            else:
+                rsp = self._moderation(content=content)
+                return rsp
+        except Exception as e:
+            logger.error("moderating failed!", e)
```
Comment (better629) [inline metagpt/provider/openai_api.py:307]: can log `e` in this format. `f"moderating failed! {e}"` or others.

**Comment by better629** [comment]
LGTM

**Comment by shenchucheng** [comment]
LGTM

**Comment by shenchucheng** [comment]
LGTM

**Comment by shenchucheng** [comment]
LGTM

**Pull request merged**